### PR TITLE
memoize getShareUrl function

### PR DESCRIPTION
**Pull request by JakeChampion**

### CODE REVIEW

File: src/js/share.js [hunk 1]
```diff
@@ -15,7 +15,24 @@
 		email: 'mailto:?subject=See this article on FT.com&body={{title}}%0A{{url}}'
 	};
 
-	const shareUrlPromises = {};
+	const getShareUrl = memoize(function(serviceUrl, maxShares, context) {
+
+		maxShares = maxShares || 1;
+		context = context || 0;
+
+		return fetch(serviceUrl + '/generate' +
```
Comment (seanmtracey) [inline src/js/share.js]: This needs to be config.serviceURL.

File: src/js/share.js [hunk 2]
```diff
@@ -40,6 +57,23 @@
 
 	}
 
+	function getRemainingNumberOfTokensForUser(serviceURL){
+
+		return fetch(serviceURL + '/remainingamount',
```
Comment (seanmtracey) [inline src/js/share.js]: See above.

File: src/js/share.js [hunk 3]
```diff
@@ -388,7 +376,7 @@
 
 		if (tokenTimeout === undefined) {
 			tokenTimeout = setTimeout(function () {
-				getShareUrl(1, 1)
+				getShareUrl(this.config.serviceURL, 1, 1)
```
Comment (seanmtracey) [inline src/js/share.js]: Getting an error that it can't find 'this.config' when run in FF

File: src/js/share.js (hunk 1 above)
Comment (JakeChampion) [inline src/js/share.js]: As this function is not a part of the prototype, it does not have access to `config`, instead it takes `serviceUrl` as the first parameter.

File: src/js/share.js (hunk 2 above)
Comment (JakeChampion) [inline src/js/share.js]: It's taken as the first parameter.

File: src/js/share.js [hunk 4]
```diff
@@ -388,7 +380,7 @@
...
```
Comment (seanmtracey) [inline src/js/share.js]: This is broken.

**Pull request merged**